test/csuite/wt12945_flags_eviction_prefetch/main.c

```diff
@@ -0,0 +1,217 @@
+        /* Do one insertion */
+        set_key(cursor, record_idx);
+        set_value(opts, cursor, record_idx);
+        testutil_check(wt_session->begin_transaction(wt_session, "isolation=snapshot"));
+        testutil_check(cursor->insert(cursor));
+        testutil_check(wt_session->commit_transaction(wt_session, NULL));
+        if (record_idx % (10 * WT_THOUSAND) == 0) {
+            printf("eviction thread: Warm-up: insert key=%" PRIu64 ", value=%" PRIu64 "\n",
+              record_idx, record_idx);
+            fflush(stdout);
+        }
+    }
+    testutil_check(cursor->close(cursor));
+
+    /* Create the thread for pre-fetch and wait for it to be ready. */
+    testutil_check(pthread_create(&prefetch_thread_id, NULL, thread_do_prefetch, opts));
+
+    for (;; __wt_yield()) {
+        WT_ACQUIRE_READ_WITH_BARRIER(ready_counter_local, ready_counter);
+        if (ready_counter_local >= 1)
+            break;
+    }
+
+    /* Loop updating documents and triggering eviction. */
+    testutil_check(
+      wt_session->open_cursor(wt_session, opts->uri, NULL, "debug=release_evict", &cursor));
+    for (record_idx = FIRST_RECORD_TO_CHANGE; record_idx < FIRST_RECORD_TO_CHANGE + NUM_EVICTION;
+         ++record_idx) {
+        /* Do one update */
+        set_key(cursor, record_idx);
+        set_value(opts, cursor, 2 * record_idx);
+        testutil_check(wt_session->begin_transaction(wt_session, NULL));
+        testutil_check(cursor->update(cursor));
+        testutil_check(wt_session->commit_transaction(wt_session, NULL));
+        if (record_idx % (10 * WT_THOUSAND) == 0) {
+            printf("eviction thread: Updates: update key=%" PRIu64 ", value=%" PRIu64 "\n",
+              record_idx, 2 * record_idx);
+            fflush(stdout);
+        }
+        /* Force eviction: Makes use of debug.release_evict. */
+        cursor->reset(cursor);
+    }
+
+    testutil_check(cursor->close(cursor));
+    testutil_check(wt_session->close(wt_session, NULL));
+
+    testutil_check(pthread_join(prefetch_thread_id, NULL));
+
+    testutil_cleanup(opts);
+    return (EXIT_SUCCESS);
+}
+
+/*
+ * thread_do_prefetch --
+ *     Read to trigger pre-fetch.
+ */
+void *
+thread_do_prefetch(void *arg)
+{
+    TEST_OPTS *opts;
+    WT_CONNECTION *conn;
+    WT_CURSOR *cursor;
+    WT_SESSION *wt_session;
+    uint64_t idx;
+    uint64_t value;
+
+    opts = (TEST_OPTS *)arg;
+    conn = opts->conn;
+
+    printf("Running prefetch thread\n");
+
+    testutil_check(conn->open_session(conn, NULL, NULL, &wt_session));
+    testutil_check(wt_session->open_cursor(wt_session, opts->uri, NULL, NULL, &cursor));
+
+    /* Tell the eviction thread that the prefetch thread is ready. */
+    (void)__wt_atomic_add64(&ready_counter, 1);
+
+    /* Read to trigger prefetch */
+    for (idx = 0; idx < FIRST_RECORD_TO_CHANGE + NUM_EVICTION; ++idx) {
+        set_key(cursor, idx);
+        testutil_check(wt_session->begin_transaction(wt_session, "isolation=snapshot"));
```

Comment:
Instead of doing this I'd just cursor->next() through the tree to trigger prefetch. I'm not sure if this method actually will.

Reply:
Done